server/src/db_manager.py

```diff
@@ -0,0 +1,197 @@
+import sqlite3
+import os
+
+class DatabaseManager(object):
+    def check_db_exists(self):
+        """check if a database file already exists"""
+        from os.path import isfile
+
+        if not isfile(self.db_name):
+            return False
+        else: return True
+
+    def init_db(self, filename):
+        self.db_name = filename
+        if(self.check_db_exists()):
+            print("Existing SQL database detected!")
+        else:
+            print("No SQL database detected! Creating...")
+            try:
+                conn = sqlite3.connect(self.db_name)
+                c = conn.cursor()
+
+                c.execute("""CREATE TABLE `users` (
+                            `username`	TEXT NOT NULL,
+                            `password`	TEXT NOT NULL,
+                            unique (username) on conflict ignore 
+                        );""")
+
+                c.execute("""CREATE TABLE `payments` (
+                            `bill_id`	INT NOT NULL,
+                            `payee_name`	TEXT NOT NULL,
+                            `amount_owed`	float NOT NULL,
+                            `is_paid` bit NOT NULL
+                        );""")
+
+                c.execute("""CREATE TABLE `bills` (
+                            `bill_id`	INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+                            `name`	TEXT NOT NULL,
+                            `username`	TEXT NOT NULL,
+                            `timestamp`	TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT NULL,
+                            `total`	float NOT NULL,
+                            `is_paid` bit NOT NULL
+                        );""")
+
+                
+            except Exception as e:
+                print(e)
```

Comment:
COMMENT: Should we print to the terminal window as done here, or should we have a log file, and log all errors? Additionally, Should the error here be thrown and handled by the caller?

Reply:
If we add a proper log file, it should be addressed in a separate issue.
As for error throwing, agreed this is not ideal because the calling function does not know of the success of the function.

I'm not sure about leaving exception handling up to the caller either because that makes the calling code quite messy. Maybe return a boolean with the success?

Reply:
Agreed that a proper log file should be a separate issue if we choose to do that. Also agree that boolean with success is a good idea in order to minimize calling complexity